framework/web/UrlRule.php

```diff
@@ -158,7 +158,8 @@ public function init()
             $this->name = $this->pattern;
         }
 
-        $this->pattern = trim($this->pattern, '/');
+        // don't trim left slashes if more then one slash at the beginning of pattern
+        $this->pattern = preg_replace('/(^(\/)(?!\/)|(\/$))/', '', $this->pattern);
```

Comment:
It does not look right. IMO `////example.com/home` should be treated as `example.com` as hostname and '/home' as path. Extra slashes should be trimmed.